File: app/src/main/java/com/bubelov/coins/ui/activity/SettingsActivity.java
package com.bubelov.coins.ui.activity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.ActivityOptionsCompat;
import android.support.v7.widget.Toolbar;

import com.bubelov.coins.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * @author Igor Bubelov
 */

public class SettingsActivity extends AbstractActivity {
    @BindView(R.id.toolbar)
    Toolbar toolbar;

    public static void start(Activity activity) {
        Intent intent = new Intent(activity, SettingsActivity.class);
        activity.startActivity(intent, ActivityOptionsCompat.makeSceneTransitionAnimation(activity).toBundle());
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);
        ButterKnife.bind(this);
        toolbar.setNavigationOnClickListener(v -> supportFinishAfterTransition());
    }
}

File: app/src/main/java/com/bubelov/coins/ui/activity/SettingsActivity.kt
package com.bubelov.coins.ui.activity

import android.app.Fragment
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import com.bubelov.coins.R
import dagger.android.AndroidInjection
import dagger.android.HasFragmentInjector

import kotlinx.android.synthetic.main.activity_settings.*
import dagger.android.DispatchingAndroidInjector
import javax.inject.Inject

/**
 * @author Igor Bubelov
 */

class SettingsActivity : AppCompatActivity(), HasFragmentInjector {
    @Inject internal lateinit var fragmentInjector: DispatchingAndroidInjector<Fragment>

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        toolbar.setNavigationOnClickListener { supportFinishAfterTransition() }
    }

    override fun fragmentInjector() = fragmentInjector

    companion object {
        fun newIntent(context: Context) = Intent(context, SettingsActivity::class.java)
    }
}